### DISCUSSION TITLE
Using graph-explorer to visualize a graph hosted on a local docker container running gremlin server

### DISCUSSION THREAD

**Discussion by miguelmserrano101** [aws/graph-explorer#159] [Q&A]
Hi, I would like to run the graph explorer to visualize a graph hosted on a local gremlin server. The server resides within a docker image (tinkerpop/gremlin-server) and can be accessed via 

localhost:8181/gremlin

I can't figure out what to put under the hostname field
docker run -it --rm  -p 80:80 -p 443:443 --env HOST={hostname-or-ip-address}  graph-explorer

may you provide some guidance on how to fill in the blank or if I'm attempting something that is not possible

**Comment by triggan** [comment]
If you're using Graph Explorer from your local machine and connecting to it via `https://localhost/explorer`, then you would use `localhost` as the hostname.

**Reply by miguelmserrano101** [reply]
Thank you for the response. The ui for explorer does show up. However, when referencing gremlin server it gets hung up. The site keep saying "synchronizing". What does this behavior imply? It also prevents any interaction with the ui.

**Reply by triggan** [reply]
Gremlin Server does not have a means by which it can store schema/data models. As such, queries need to be executed to derive the graph schema that Explorer can then use in order to provide visualization capabilities. On larger graphs, then can take a long time to execute. For Neptune, we've addressed this on Neptune versions that support the newer Summary API [1] which can provide enough detail to avoid needing to derive schema from queries.

I thought we already had an issue submitted for allowing users to manually declare their graph schema, but I can't seem to find it. I just created a new one. Please +1 this feature request when appropriate. [2]

[1] https://aws.amazon.com/about-aws/whats-new/2023/03/amazon-neptune-graph-summary-api/
[2] https://github.com/aws/graph-explorer/issues/160
